### ISSUE TITLE
Feature request: DynamoDB support 

### ISSUE THREAD

**Issue by pasku** [TablePlus/TablePlus#666]
Just throwing it out there.

**Comment by Wilddata** [comment]
Using DynamoDb-GUI-Client as well, better than nothing but it's very limited. We really need to be added to TablePlus. Any ETA??

**Comment by nosykretts** [comment]
@Wilddata AWS just released "NoSQL Workbench for Amazon DynamoDB" available on macos and windows.

**Comment by QuentinDela** [comment]
You can also use https://dynobase.dev/

**Comment by Wilddata** [comment]
I don't think I'm gonna try that one. The site is locked by Chrome for security reasons...

**Comment by natebrunette** [comment]
I couldn't get nosql workbench to connect to my local instance. I'd still +1 this feature.

**Comment by brianpham** [comment]
Would love for this feature too :) +1

**Comment by rferro** [comment]
+1

**Comment by outaTiME** [comment]
🙏

**Comment by p0wl** [comment]
👍 any updates if this is planned or beeing worked on?

**Comment by bobbyrc** [comment]
+1

**Comment by theclaw37** [comment]
+1

**Comment by erashdan** [comment]
+1

**Comment by Tiketti** [comment]
By commenting "+1", you are triggering email notifications to people
watching this issue and creating noise. I kindly ask you to please be
considerate and e.g. use the emoji reactions to single posts.

**Comment by lannsjo** [comment]
Support for DynamoDb would be awsome, there is no good alternative out there at the moment.

**Comment by alsantos123** [comment]
Yeah, I will love use more TablePlus whether it could did that.

**Comment by aedenmurray** [comment]
+1

**Comment by dperconti** [comment]
Does anyone know if this feature work is planned?

**Comment by tamvm** [comment]
Hope to have this soon, 


(note that AWS has Dynamo running locally already https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html)

**Comment by everdrone** [comment]
This feature would be amazing 
+1

**Comment by batFormat** [comment]
+1

**Comment by romcok** [comment]
Please 🙏 Dynamo is a document key value store, easier for queries than mongo.

**Comment by GB-BABA** [comment]
+1

**Comment by huyphams** [comment]
We're working at this driver, but it is not ready for the public beta.

**Comment by helman** [comment]
any ETA for public beta or release? I'm mainly working with dynamodb and my tableplus app is kind of sitting idle at the moment.

**Comment by pregnantboy** [comment]
any ETA?

**Comment by mrded** [comment]
Someone had to notice the PR number 🔥🐐 ⭐

**Comment by theoboldalex** [comment]
Would love to see Dynamo DB support. Any ideas if/when this will be available?

**Comment by AlanSyue** [comment]
+1

**Comment by perriea** [comment]
Any news ? :(

**Comment by capt4ce** [comment]
Waiting for support of this feature as well

**Comment by dndeus** [comment]
People from the TablePlus team kindly request an update on this since many of us are paying customers, so an update on this wouldn't hurt wether there is progress or not

**Comment by bw-sonph** [comment]
+1

**Comment by romaindefo** [comment]
+1

**Comment by GB-BABA** [comment]
+1

**Comment by shiveenp** [comment]
hello, this would be really good, especially since the big JB just added it as well https://www.jetbrains.com/help/idea/dynamodb.html but I prefer and pay for the Table Plus UI.

**Comment by sandrodz** [comment]
Yes, table plus is awesome. Make dynamodb a premium feature for paid users :)

**Comment by ned-kelly** [comment]
+1

**Comment by TuralAsgar** [comment]
+1

**Comment by morvaiz** [comment]
+1

**Comment by fishcharlie** [comment]
Can the TablePlus maintainers please lock this thread? Everyone constantly replying with +1 is extremely unproductive and just creates unnecessary spam for the people who are subscribed to this issue waiting for actual updates.

If you want to express your interest in this issue consider adding a 👍 reaction to the original post.

**Comment by bernier** [comment]
I guess it's because the issue is 666

**Comment by shelby-carter** [comment]
+10000000000

**Comment by marvelph** [comment]
+1

**Comment by riggerz29** [comment]
+1

**Comment by Seikida** [comment]
+1

**Comment by dperconti** [comment]
![boo](https://c.tenor.com/g-BQe4TkUBMAAAAC/tenor.gif)

**Comment by r-moore** [comment]
This was the #1 feature request in terms of github 'votes'. Did the team investigate and deem it not possible or something? Technical issue, licensing problem, too many other priorities?

**Comment by huyphams** [comment]
We may add it soon https://docs.aws.amazon.com/sdk-for-swift/latest/developer-guide/swift_dynamodb_code_examples.html

**Comment by huyphams** [comment]
Working on it rn, will update this thread as soon as we have a test build.

**Comment by huyphams** [comment]
<img width="1121" height="776" alt="Image" src="https://github.com/user-attachments/assets/96f04db8-a524-4a1f-af80-11a1ffb9c304" />

Hi, which DynamoDB feature you want to see in TablePlus?

**Comment by Fishy49** [comment]
> Hi, which DynamoDB feature you want to see in TablePlus?

Hi @huyphams !

This list is not comprehensive by any means but here are some things I'd like to see:

- Ability to view a table's schema in JSON format
- Support for querying (filtering) by defined indices - TablePlus should detect and allow dropdowns for existing indices on a table
- Warnings if a query/filter is going to result in a table scan (e.g. the fields queried are not indexed)
- Ability to export a table to a JSON/CSV file (will cause a table scan, but super useful)
- Ability to import to a table from a JSON/CSV file
- Ability to create a table by importing a JSON definition
- Ability to edit records inline like you can with current drivers

**Comment by huyphams** [comment]
From this patch, we're working on the DynamoDB 
https://www.dropbox.com/scl/fi/mpnhecs75ruoqe3442fpa/TablePlus.dmg?rlkey=594kh1wtsg85zmqcslim6jitj&dl=0

Done:
- View tables
- Filter table
Working:
- Update rows
- View schemas
- Import/Export

**Comment by bernier** [comment]
Truncate table would be a nice feature as well

**Comment by marvelph** [comment]
Should I report the issue where only one table is displayed as a separate bug?
